src/ucp/tag/rndv.c

```diff
@@ -195,6 +246,28 @@ static void ucp_rndv_complete_rndv_get(ucp_request_t *rndv_req)
     ucp_rndv_send_ats(rndv_req, rndv_req->send.rndv_get.remote_request);
 }
 
+static void ucp_rndv_complete_rndv_put(ucp_request_t *rndv_req)
+{
+    ucp_request_t *sreq = rndv_req->send.rndv_put.sreq;
+
+    ucs_assertv(rndv_req->send.state.offset == rndv_req->send.length,
+                "rndv_req=%p offset=%zu length=%zu", rndv_req,
+                rndv_req->send.state.offset, rndv_req->send.length);
+
+    ucs_trace_data("ep: %p rndv put completed", rndv_req->send.ep);
+
+    UCS_PROFILE_REQUEST_EVENT(sreq, "complete_rndv_get", 0); // TODO
+    ucp_request_complete_send(sreq, UCS_OK);
+
+    if (rndv_req->send.rndv_put.rkey_bundle.rkey != UCT_INVALID_RKEY) {
+        uct_rkey_release(&rndv_req->send.rndv_put.rkey_bundle);
+    }
+
+    ucp_rndv_rma_request_send_buffer_dereg(rndv_req);    
+
+    ucp_rndv_send_atp(rndv_req, rndv_req->send.rndv_put.remote_request);
```

Comment:
move this above ucp_rndv_rma_request_send_buffer_dereg(), so that ctrl pkt will be send out sooner